src/PIL/TgaImagePlugin.py

```diff
@@ -151,11 +153,23 @@ def _save(im, fp, filename):
     except KeyError:
         raise IOError("cannot write mode %s as TGA" % im.mode)
 
-    rle = im.encoderinfo.get("rle", False)
-
+    if "rle" in im.encoderinfo:
+        rle = im.encoderinfo["rle"]
+    else:
+        compression = im.encoderinfo.get("compression",
+                                         im.info.get("compression"))
+        rle = compression == "tga_rle"
     if rle:
         imagetype += 8
 
+    id_section = im.encoderinfo.get("id_section",
+                                    im.info.get("id_section", ""))
+    idlen = len(id_section)
+    if idlen > 255:
+        idlen = 255
```

Comment:
Let's `snake_case` this variable.

Reply:
Okay, I've added a commit. Note that the commit also changes the `idlen` variable in `_open`, for consistency.